### ISSUE TITLE
HW08 is ready for grading

### ISSUE THREAD

**Issue by ZimingY** [ZimingY/STAT545-hw-YIN-ZIMING#7]
You can find my 'readme' [here](https://github.com/ZimingY/STAT545-hw-YIN-ZIMING/blob/master/Hw8/ReadMe.md).

Get to my online shiny app [here](https://ziming123.shinyapps.io/my_bc_liquor_app/).

**Comment by mattsada** [comment]
Hi @ZimingY and thanks for inviting me! Here is my peer review of your efforts in this assignment and I wish you the best of luck with all of your future assignments. Overall, I think you did a great job with this assignment. You did everything that was required from you and your app runs smoothly without any major errors. 

- I like your [README](https://github.com/ZimingY/STAT545-hw-YIN-ZIMING/tree/master/Hw8) file. It gives the reviewer a great overview of the features that you've implemented in your final app and also, you state the origin of your input data, great!

- However, there seems to be missing a "reflections" section and it would be nice to hear your thoughts on developing this app. Did you have any major/minor struggles and if you did, how did you overcome these struggles?

- Also, it would be nice to include some information within your UI in your app in regards of the origin of the data used for this app. 

- Indeed a cool image was successfully imbedded in your app, awesome!

- The DT package seems to do its job in making the tables interactive.

- The number of results are displayed at the bottom left corner of your table.

- Search function for multiple alcohol types are also working properly.

- Finally, the download button seems to work perfectly.

To summarize, all of your implemented functions seems to be working perfectly and so is your app. Your code is neatly formatted and concise which helps in navigating through your code chunks.  Keep up the good work and I'll see you in class. Cheers!

**Comment by mylinhthibodeau** [comment]
Dear @ZimingY,

Sorry for the late review, I have been sick with a cold recently and somehow, that one slipped through. Here is a funny one: I also performed the wrong peer review. Oh well, my mom used to say all the time: *Mieux vaut en rire qu'en pleurer* (which means, better to laugh about it than cry about it, and that's the way I will go with it ;) !

## Peer Review

Excellent work, your app fulfill all the requirements:

1. Split into split into a `ui.R` and a `server.R` 
2. Deployed online on shinyapps.io [here](https://ziming123.shinyapps.io/my_bc_liquor_app/)
3. Has a README.md file which is informative and details your process. 
4. Code of Shiny app included in your repository 
5. Acknowledge the source of dataset clearly and extensively

You also explored these additional features:

* Used `DT::dataTableOutput()` to render a searchable table.
* Used `tabsetPanel()` to make two panels (plot and table)
* Used the `downloadButton()` to make data table available to user ! 
* Added image with `img(src = "alch.jpg",width = "100%")`
* Used `reactive({})` to filter data according to user input. 
* Used `renderText({})` to output the number of options found. 
* Used `tags$a()` to provide hyperlink in Shiny App. 
* Used `selectInput()` with `multiple = TRUE` to allow multiple user input. 

You reflected on your process in your README file, and I also like that your `ui.R` and `server.R` files are neat and tidy. One potential idea could be to add some comments to your code directly, so that when you read it back, it will be very easy to understand.

### Additional tips

Your homework fulfill all the requirements, but I thought I should also provide some helpful comments and therefore, here are some little tips you might find helpful for your learning. 

* When splitting your Shiny App into a `ui.R` and a `server.R`, you don't have to put the `ui <-` and `server <-` portions, you can start directly with `fluidPage(` and `function(input, output) {` respectively. 
* I would suggest to provide the reference of the original dataset for the BC Liquor Store data, just for tracking purposes in the future. 
* One potential visual trick could be to add colors for each Product type in the graph (so that the user knows which Products are more represented).

## Great job on this homework !! Good luck for the rest of the course :)

Warm regards,
My Linh

**Comment by ksedivyhaley** [comment]
**App uses ui.R and server.R file:** Yes
**Deployed online:** Yes
**Course repo includes README, code:** Yes
**Acknowledge datset source:** OK (see comments)
**New app or 3+ added features:** Yes (5 features)
**Reflection:**  No

**Comments:**

- You do not need app.R if you are using ui.R and server.R, and can remove the file from your repo.
- Dean's not the original source of the data – you want to cite the place he got the data from.
- When you return one result, text says “We've found 1 options for you”. It would be nice to print “1 option”.
- There is an error when you have no product type selected. It would be nice to fix this bug.
- We'd like to see more details on how you found the process of coding.

*Your mark will be distributed later. If you would like more feedback, please feel free to message me on slack.*
